WIP: Resolve CVE 2015 9284 with POST to azure

awead opened the pull request:
Resolves the security issue with CVE 2015 9284 by allowing only POST methods to the Azure omni auth service.

Note: This works, but the JS-based tests fail.

awead commented:
@mtribone here's what I'm thinking. This changes the text of the link, however, the screen reader doesn't seem to pick it up, even with using aria. Maybe JAWS would?

Before:
![before](https://cloud.githubusercontent.com/assets/312085/23760905/20430e1e-04bf-11e7-8542-1ca7ad878852.png)

After:
![after](https://cloud.githubusercontent.com/assets/312085/23760912/25caa1a8-04bf-11e7-88fa-1ddca6b9e036.png)

awead commented:
@mtribone  BTW, I initially tried a span, but it didn't seem to pick that up either.

mtribone commented:
@awead It's so simple it is wonderful. Tried to use the big JS hammer..._puts head into hands_

mtribone commented on app/assets/javascripts/sufia/save_work/checklist_item.es6:
```diff
@@ -0,0 +1,19 @@
+  check() {
+    this.element.removeClass('incomplete')
+    this.element.addClass('complete')
+    this.element.children('a').text(this.element.data("complete"))
```
I tried to put the `<span>` here and it just repeated the content 5 times as I filled out the form.

awead commented:
@mtribone don't be hard on yourself 😉

awead commented:
@mtribone ok, this has changed a bit under the hood, but the concept is the same: change the links. I'm no longer concerned that the user hear the changes immediately, instead, the reader will announce the text when the user gets to it in the course of moving through the form.

Also, the links were updated with some JS to make them active the tabs.

awead committed:
WIP: Resolve CVE 2015 9284 with POST to azure

Resolves the security issue with CVE 2015 9284 by allowing only POST
methods to the Azure omni auth service.

awead commented:
It looks like there's a forthcoming 2.0 release that will address this. https://github.com/omniauth/omniauth/pull/1010
